internal/controller/apiextensions/definition/handlers.go

```diff
@@ -35,90 +33,106 @@ import (
-	return handler.Funcs{
-		UpdateFunc: func(ctx context.Context, ev kevent.UpdateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
-			xrGVK := of
-			cdGVK := ev.ObjectNew.GetObjectKind().GroupVersionKind()
-			key := refKey(ev.ObjectNew.GetNamespace(), ev.ObjectNew.GetName(), cdGVK.Kind, cdGVK.GroupVersion().String())
-
-			composites := kunstructured.UnstructuredList{}
-			composites.SetGroupVersionKind(xrGVK.GroupVersion().WithKind(xrGVK.Kind + "List"))
-			if err := c.List(ctx, &composites, client.MatchingFields{compositeResourcesRefsIndex: key}); err != nil {
-				log.Debug("cannot list composite resources related to a composed resource change", "error", err, "gvk", xrGVK.String(), "fieldSelector", compositeResourcesRefsIndex+"="+key)
-				return
-			}
+// SelfMapFunc returns a MapFunc that maps an object to itself for reconciliation.
+// This is equivalent to handler.EnqueueRequestForObject but as a MapFunc.
+func SelfMapFunc() handler.MapFunc {
+	return func(_ context.Context, obj client.Object) []reconcile.Request {
+		return []reconcile.Request{{
+			NamespacedName: types.NamespacedName{
+				Name:      obj.GetName(),
+				Namespace: obj.GetNamespace(),
+			},
+		}}
+	}
+}
 
-			// queue those composites for reconciliation
-			for _, xr := range composites.Items {
-				log.Debug("Enqueueing composite resource because composed resource changed", "name", xr.GetName(), "namespace", xr.GetNamespace(), "cdGVK", cdGVK.String(), "cdName", ev.ObjectNew.GetName())
-				q.Add(reconcile.Request{NamespacedName: types.NamespacedName{Name: xr.GetName(), Namespace: xr.GetNamespace()}})
-			}
-		},
+// CompositeResourcesMapFunc returns a MapFunc that maps composed resources to affected XRs.
+func CompositeResourcesMapFunc(of schema.GroupVersionKind, c client.Reader, log logging.Logger) handler.MapFunc {
+	return func(ctx context.Context, obj client.Object) []reconcile.Request {
+		cdGVK := obj.GetObjectKind().GroupVersionKind()
+		key := refKey(obj.GetNamespace(), obj.GetName(), cdGVK.Kind, cdGVK.GroupVersion().String())
+
+		composites := kunstructured.UnstructuredList{}
+		composites.SetGroupVersionKind(of.GroupVersion().WithKind(of.Kind + "List"))
+		if err := c.List(ctx, &composites, client.MatchingFields{compositeResourcesRefsIndex: key}); err != nil {
+			log.Debug("cannot list composite resources related to a composed resource change", "error", err, "gvk", of.String(), "fieldSelector", compositeResourcesRefsIndex+"="+key)
+			return nil
+		}
+
+		requests := make([]reconcile.Request, len(composites.Items))
+		for i, xr := range composites.Items {
+			log.Debug("Mapping composite resource because composed resource changed",
+				"namespace", xr.GetNamespace(),
+				"name", xr.GetName(),
+				"cdGVK", cdGVK.String(),
+				"cdName", obj.GetName(),
+			)
+			requests[i] = reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&xr)}
```

Comment:
i think this is the equivalent line from https://github.com/crossplane/crossplane/pull/6780, which would be using the namespace inside client.ObjectKeyFromObject, so I think that functionality is still intact 🙏